source/Nevermore/Advanced/ReaderStrategies/Primitives/PrimitiveReaderStrategy.cs

```diff
@@ -23,7 +23,8 @@ public bool CanRead(Type type)
                 || type.IsPrimitive
                 || type.IsValueType
                 || type.IsEnum
-                || (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>));
+                || (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>))
+                || configuration.TypeHandlers.Resolve(type) != null;
```

Comment:
nasty, this needs to match with `ExpressionHelper.GetValueFromReaderAsType`, not obvious